sim-administration/sim-batch-management/es2plus/es2plus.go

```diff
@@ -0,0 +1,401 @@
+	Iccid                     string `json:"iccid"`
+	LockFlag                  bool   `json:"lockFlag"`
+}
+
+
+//
+//  Profile reset invocation
+//
+
+type ES2PlusRecoverProfileRequest struct {
+    Header          ES2PlusHeader  `json:"header"`
+    Iccid           string         `json:"iccid"`
+    ProfileStatus   string         `json:"profileStatus"`
+ }
+
+type ES2PlusRecoverProfileResponse struct {
+	Header    ES2PlusResponseHeader  `json:"header"`
+}
+
+
+
+//
+//  Cancel order  invocation
+//
+
+type ES2PlusCancelOrderRequest struct {
+    Header          ES2PlusHeader  `json:"header"`
+    Iccid           string         `json:"iccid"`
+    FinalProfileStatusIndicator   string         `json:"finalProfileStatusIndicator"`
+ }
+
+type ES2PlusCancelOrderResponse struct {
+	Header    ES2PlusResponseHeader  `json:"header"`
+}
+
+//
+//  Download order invocation
+//
+
+type ES2PlusDownloadOrderRequest struct {
+    Header          ES2PlusHeader  `json:"header"`
+    Iccid           string         `json:"iccid"`
+    Eid             string         `json:"eid,omitempty"`
+    Profiletype     string         `json:"profiletype,omitempty"`
+}
+
+type ES2PlusDownloadOrderResponse struct {
+	Header    ES2PlusResponseHeader  `json:"header"`
+	Iccid     string         `json:"iccid"`
+}
+
+//
+// ConfirmOrder invocation
+//
+
+type ES2PlusConfirmOrderRequest struct {
+	Header            ES2PlusHeader  `json:"header"`
+	Iccid             string         `json:"iccid"`
+	Eid               string         `json:"eid,omitempty"`
+	MatchingId        string         `json:"matchingId,omitempty"`
+	ConfirmationCode  string         `json:"confirmationCode,omitempty"`
+	SmdpAddress       string         `json:"smdpAddress,omitempty"`
+	ReleaseFlag       bool           `json:"releaseFlag"`
+}
+
+type ES2PlusConfirmOrderResponse struct {
+	Header         ES2PlusResponseHeader  `json:"header"`
+	Iccid          string         `json:"iccid"`
+	Eid            string         `json:"eid,omitempty"`
+    MatchingId     string         `json:"matchingId,omitempty"`
+    SmdpAddress    string         `json:"smdpAddress,omitempty"`
+}
+
+
+//
+//  Generating new ES2Plus clients
+//
+
+
+type Es2PlusClient struct {
+	httpClient *http.Client
+	hostport string
+	requesterId string
+	printPayload bool
+	printHeaders bool
+}
+
+func Client (certFilePath string, keyFilePath string, hostport string, requesterId string) (*Es2PlusClient) {
+    return &Es2PlusClient {
+        httpClient: newHttpClient(certFilePath, keyFilePath),
+        hostport: hostport,
+        requesterId: requesterId,
+        printPayload: false,
+        printHeaders: false,
+    }
+}
+
+func newHttpClient(certFilePath string, keyFilePath string) *http.Client {
+	cert, err := tls.LoadX509KeyPair(
+		certFilePath,
+		keyFilePath)
+	if err != nil {
+		log.Fatalf("server: loadkeys: %s", err)
+	}
+	config := tls.Config{Certificates: []tls.Certificate{cert}, InsecureSkipVerify: true}
```

Comment:
Remove `InsecureSkipVerify` or add a TODO and a good reason why this can't be fixed now.